From a5fd2ae8d4a3b2a8f7a33a4ea469ea7ee0d946ef Mon Sep 17 00:00:00 2001
From: Christian Gromm <christian.gromm@microchip.com>
Date: Mon, 4 Sep 2017 15:36:38 +0200
Subject: [PATCH 4/5] most: video: set device_caps

Signed-off-by: Christian Gromm <christian.gromm@microchip.com>
---
 driver/aim-v4l2/video.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/aim-v4l2/video.c b/aim-v4l2/video.c
index e074841..6405a03 100644
--- a/aim-v4l2/video.c
+++ b/aim-v4l2/video.c
@@ -263,6 +263,7 @@ static int vidioc_querycap(struct file *file, void *priv,
 	snprintf(cap->bus_info, sizeof(cap->bus_info),
 		 "%s", mdev->iface->description);
 
+	cap->device_caps =
 	cap->capabilities =
 		V4L2_CAP_READWRITE |
 		V4L2_CAP_TUNER |
-- 
2.7.4

